pulp_smash/tests/rpm/api_v2/test_sync_publish.py

```diff
@@ -438,3 +473,85 @@ def test_unit_integrity(self):
         for i, module in enumerate(self.rpms[1:]):
             with self.subTest(i=i):
                 self.assertEqual(self.rpms[0], module)
+
+
+class SyncOnDemandTestCase(_BaseTestCase):
+    """Assert Yum supports on-demand syncing."""
+
+    @classmethod
+    def setUpClass(cls):
+        """Create an RPM repository with a valid feed and sync it.
+
+        Do the following:
+
+        0. Reset Pulp, including the Squid cache.
+        1. Sync/publish a repository using the 'on_demand' download policy.
+        2. Download an RPM from the published repository.
+        3. Download the same RPM to ensure it is served by the cache.
+        """
+        super(SyncOnDemandTestCase, cls).setUpClass()
+
+        # Required to ensure the `locally_stored_units` is 0 before we start.
+        utils.reset_squid(cls.cfg)
+        utils.reset_pulp(cls.cfg)
+        importer_config = {
+            'feed': RPM_FEED_URL,
+            'download_policy': 'on_demand',
+        }
+        repo, tasks = _sync_and_publish_repo(
+            cls.cfg,
+            importer=importer_config
+        )
+        client = api.Client(cls.cfg, api.echo_handler)
```

Comment:
Why is `api.echo_handler` used here? It's a no-op callback; it returns server responses untouched.